components/org.wso2.carbon.deployment.automation/src/main/java/org/wso2/carbon/deployment/automation/kubernetes/KubernetesDeploymentProvider.java

```diff
@@ -186,18 +190,29 @@ public void undeploy(Deployment deployment) throws DeploymentEnvironmentExceptio
      */
     private Map<String, String> fetchResources(String path) throws DeploymentEnvironmentException {
         Map<String, String> resources = new HashMap<String, String>();
-        Map<String, Object> map = null;
+        Map<String, Object> map;
+        FileInputStream fileInputStream = null;
+
         try {
-            map = (Map<String, Object>) new Yaml().load(new FileInputStream(new File(path)));
+            fileInputStream = new FileInputStream(new File(path));
+            map = (Map<String, Object>) new Yaml().load(fileInputStream);
         } catch (FileNotFoundException e) {
-            throw new DeploymentEnvironmentException("Unable to find the profle for the given product", e);
+            throw new DeploymentEnvironmentException("Unable to find the profile for the given product", e);
+        } finally {
+            if (fileInputStream != null) {
+                try {
+                    fileInputStream.close();
+                } catch (IOException ignore) {
+                }
+            }
         }
-        if (map.get("kind").toString().toLowerCase().equals(KIND_LIST)) {
-            for (LinkedHashMap item : (List<LinkedHashMap>) map.get("items")) {
-                resources.put(item.get("kind").toString().toLowerCase(), new Yaml().dump(item));
+
+        if (map.get(KIND).toString().equalsIgnoreCase(KIND_LIST)) {
```

Comment:
Should not this be checked in the other way around to avoid NPEs ? We can anyway consider it as a best practice.